bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java

```diff
@@ -208,8 +209,13 @@ protected void doGet(@Nonnull SlingHttpServletRequest request, @Nonnull SlingHtt
         if (!handleIfModifiedSinceHeader(request, response, lastModifiedEpoch)) {
             Double quality = null;
             Integer jpegQualityInPercentage = null;
+            Boolean forceQuality = false;
             if (selectors.length > 2) {
                 String qualitySelector = selectors[selectors.length - 2];
+                if (StringUtils.startsWith(qualitySelector, "f")) {
```

Comment:
As this has an impact on performance the selector should only be considered in case it has been enabled in the content policy.